pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -727,21 +769,50 @@ func getDiskType(diskType string) (string, error) {
 	}
 }
 
-func (gce *GCECloud) DeleteDisk(diskToDelete string) error {
-	err := gce.doDeleteDisk(diskToDelete)
+func (gce *GCECloud) DeleteDisk(name string, zone string) error {
+	if zone == "" {
+		return fmt.Errorf("a zone must be specified")
+	}
+
+	err := gce.deleteDiskByName(name, zone)
+
 	if isGCEError(err, "resourceInUseByAnotherResource") {
 		return volume.NewDeletedVolumeInUseError(err.Error())
 	}
 
-	if err == cloudprovider.DiskNotFound {
+	if _, ok := err.(*DiskNotFoundError); ok {
+		return nil
+	}
+	return err
+}
+
+func (gce *GCECloud) DeleteRegionalDisk(name string) error {
+	err := gce.deleteRegionalDiskByName(name)
+
+	if isGCEError(err, "resourceInUseByAnotherResource") {
+		return volume.NewDeletedVolumeInUseError(err.Error())
+	}
+
+	if _, ok := err.(*DiskNotFoundError); ok {
 		return nil
 	}
 	return err
 }
 
 // ResizeDisk expands given disk and returns new disk size
-func (gce *GCECloud) ResizeDisk(diskToResize string, oldSize resource.Quantity, newSize resource.Quantity) (resource.Quantity, error) {
-	disk, err := gce.GetDiskByNameUnknownZone(diskToResize)
+func (gce *GCECloud) ResizeDisk(diskToResize string, zoneSet sets.String, oldSize resource.Quantity, newSize resource.Quantity) (resource.Quantity, error) {
+	var disk *GCEDisk
+	var err error
+	switch zoneSet.Len() {
+	case 0:
+		return oldSize, fmt.Errorf("zone information must be specified")
+	case 1:
+		disk, err = gce.getDiskByName(diskToResize, zoneSet.UnsortedList()[0])
+	case 2:
```

Comment:
Can we use constants for these throughout, like "regionalNumZones"

Reply:
I'm going to add something like "NumZonesSingleZone" too, to make it easier to track all places that rely on this kind of logic.